scripts/core/helpers/show-popup-hoc.tsx

```diff
@@ -0,0 +1,42 @@
+import {Placement} from '@popperjs/core';
+import {showPopup} from 'core/ui/components/popupNew';
+import React, {ReactNode} from 'react';
+
+interface IProps {
+    children(toggle: (referenceElement: HTMLElement) => void): ReactNode;
+    placement: Placement;
+    Component: React.ComponentType<{closePopup(): void}>;
+    zIndex?: number;
+    closeOnHoverEnd?: boolean;
+    onClose?: () => void;
+}
+
+export class ShowPopoverHoc extends React.PureComponent<IProps> {
```

Comment:
There's a naming convention for higher order components to start with `with`, like `WithPopover`. We can make exceptions, but conventional name sounds good to me, so maybe let's use that.

It'd be good also to have props camelCased `Component` -> `component`